### PR TITLE
Add requested citations to README

### PR DESCRIPTION
**Pull request by jmduarte**
- Add requested citations to README
- Documentation update

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,3 +56,67 @@ hls_model.build()
 #Print out the report if you want
 hls4ml.report.read_vivado_report('my-hls-test')
 ```
+
+# Citation
+If you use this software in a publication, please cite the software
+```bibtex
+@software{vloncar_2021_5680908,
+  author       = {Loncar, Vladimir and others},
```
Comment (vloncar) [inline README.md:64]: Should be "FastML team"

File: README.md [hunk 2]
```diff
@@ -56,3 +56,67 @@ hls_model.build()
...
+    number = "07",
+    pages = "P07027",
+    year = "2018"
+}
+```
+Additionally, if you use specific features developed in later papers, please cite those as well. For example, CNNs:
+```bibtex
+@article{Aarrestad:2021zos,
+    author = "Aarrestad, Thea and others",
+    title = "{Fast convolutional neural networks on FPGAs with hls4ml}",
```
Comment (vloncar) [inline README.md:94]: Hmm, while this paper has CNN in it's title, it describes a deprecated implementation that no one uses. The one people use is either [this](https://iopscience.iop.org/article/10.1088/2632-2153/ac9cb5) one or the new io_parallel one that is not published anywhere

File: README.md (hunk 1 above)
Comment (jmduarte) [reply README.md]: fixed.

File: README.md (hunk 2 above)
Comment (jmduarte) [reply README.md:93]: It's a valid point. I added the other CNN paper. At the end of the day, it's up to the user to cite whatever they want, and maybe once the new implementation is published somewhere we can point to that one instead.

**Review by nhanvtran — APPROVED** [review]

**Pull request merged**